Sources/BraintreeCoreSwift/BTAPIClientSwift.swift

```diff
@@ -1,5 +1,441 @@
+                        self.apiHTTP = BTAPIHTTP(url: apiURL, accessToken: accessToken)
+                    }
+                }
+
+                if self.http == nil {
+                    let baseURL: URL? = configuration?.json?["clientApiUrl"].asURL()
+
+                    if let clientToken = self.clientToken, let baseURL {
+                        self.http = BTHTTP(url: baseURL, authorizationFingerprint: clientToken.authorizationFingerprint)
+                    } else if let tokenizationKey = self.tokenizationKey, let baseURL {
+                        self.http = BTHTTP(url: baseURL, tokenizationKey: tokenizationKey)
+                    }
+                }
+
+                if self.graphQLHTTP == nil {
+                    let graphQLBaseURL: URL? = self.graphQLURL(forEnvironment: configuration?.environment ?? "")
+
+                    if let clientToken = self.clientToken, let graphQLBaseURL {
+                        self.graphQLHTTP = BTGraphQLHTTP(url: graphQLBaseURL, authorizationFingerprint: clientToken.authorizationFingerprint)
+                    } else if let tokenizationKey = self.tokenizationKey, let graphQLBaseURL {
+                        self.graphQLHTTP = BTGraphQLHTTP(url: graphQLBaseURL, tokenizationKey: tokenizationKey)
+                    }
+                }
+            }
+
+            completion(configuration, nil)
+        }
+    }
+
+    /// Fetches a customer's vaulted payment method nonces.
+    /// Must be using client token with a customer ID specified.
+    ///  - Parameter completion: Callback that returns either an array of payment method nonces or an error
+    public func fetchPaymentMethodNonces(completion: @escaping ([BTPaymentMethodNonce]?, Error?) -> Void) {
+        fetchPaymentMethodNonces(false, completion: completion)
+    }
+
+    /// Fetches a customer's vaulted payment method nonces.
+    /// Must be using client token with a customer ID specified.
+    ///  - Parameters:
+    ///   - defaultFirst: Specifies whether to sort the fetched payment method nonces with the default payment method or the most recently used payment method first
+    ///   - completion: Callback that returns either an array of payment method nonces or an error
+    public func fetchPaymentMethodNonces(_ defaultFirst: Bool, completion: @escaping ([BTPaymentMethodNonce]?, Error?) -> Void) {
+        if clientToken == nil {
+            completion(nil, BTAPIClientError.notAuthorized)
+            return
+        }
+
+        let defaultFirstValue: String = defaultFirst ? "true" : "false"
+        let parameters: [String: Any] = [
+            "default_first": defaultFirstValue,
+            "session_id": metadata?.sessionID ?? ""
+        ]
+
+        get("v1/payment_methods", parameters: parameters) { body, response, error in
+            if let error {
+                completion(nil, error)
+                return
+            }
+
+            var paymentMethodNonces: [BTPaymentMethodNonce]? = []
```

Comment:
Could this be non-optional? Or is there a chance it will be set to nil?

Reply:
It can be non-optional. Updating.